src/main.cpp

```diff
@@ -5914,22 +5910,29 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
                 // updated, reject messages go out, etc.
                 MarkBlockAsReceived(resp.blockhash); // it is now an empty pointer
                 fBlockRead = true;
+                // mapBlockSource is only used for sending reject messages and DoS scores,
```

Comment:
Race if multiple nodes send you something near the same time you might send the DoS/Reject to the "wrong" node, of course we dont really care because we'll get to them next time they send us a block (and someone will still get it).